File: solutions/java/bird-watcher/1/src/main/java/BirdWatcher.java
import java.util.Arrays;

class BirdWatcher {
    private final int[] birdsPerDay;

    public BirdWatcher(int[] birdsPerDay) {
        this.birdsPerDay = birdsPerDay.clone();
    }

    public int[] getLastWeek() {
        return birdsPerDay.length < 1 ? new int[] {0} 
            : Arrays.copyOfRange(birdsPerDay, birdsPerDay.length - 7, birdsPerDay.length);
    }

    public int getToday() {
        return birdsPerDay.length == 0 ? 0 : birdsPerDay[birdsPerDay.length - 1];
    }

    public void incrementTodaysCount() {
        birdsPerDay[birdsPerDay.length - 1] = getToday() + 1;
    }

    public boolean hasDayWithoutBirds() {
        return Arrays.stream(birdsPerDay).anyMatch(it -> it == 0);
    }

    public int getCountForFirstDays(int numberOfDays) {
        return (int) Arrays.stream(birdsPerDay).limit(Math.min(numberOfDays, birdsPerDay.length)).sum();
    }

    public int getBusyDays() {
        return (int) Arrays.stream(birdsPerDay).filter(it -> it > 4).count();
    }
}

File: csharp/bird-watcher/BirdWatcher.cs
class BirdCount
{
    private int[] _birdsPerDay;

    public BirdCount(int[] birdsPerDay)
    {
        _birdsPerDay = birdsPerDay;
    }

    public static int[] LastWeek() => [0, 2, 5, 3, 7, 8, 4];

    public int Today() => _birdsPerDay[^1];

    public void IncrementTodaysCount() => _birdsPerDay[^1] += 1;

    public bool HasDayWithoutBirds() => _birdsPerDay.Any(x => x == 0);

    public int CountForFirstDays(int numberOfDays) => _birdsPerDay.Take(numberOfDays).Sum();

    public int BusyDays() => _birdsPerDay.Count(x => x >= 5);
}
